extensions/wikia/TemplateClassification/scripts/TemplateClassificationModal.js

```diff
@@ -107,8 +115,40 @@ function ($, mw, loader, nirvana, labeling) {
 	 */
 	function processInstance(modalInstance) {
 		/* Submit template type edit form on Done button click */
-		modalInstance.bind('done', function runSave() {
+		modalInstance.bind('done', function runSave(e) {
 			processSave(modalInstance);
+
+			// Track - primary-button click
+			tracker.track({
+				trackingMethod: 'both',
+				category: 'template-classification-dialog',
+				action: 'primary-button',
+				label: $(e.currentTarget).text()
+			});
+		});
+
+		modalInstance.bind('close', function () {
+			// Track - close TC modal
+			tracker.track({
+				trackingMethod: 'both',
+				category: 'template-classification-dialog',
+				action: 'close',
+				label: 'close-event'
+			});
+		});
+
+		modalInstance.bind('option-select', function(e) {
+			var $input = $(e.currentTarget).find('input:radio');
+
+			$input.attr('checked', 'checked');
+
+			// Track - click to change a template's type
+			tracker.track({
+				trackingMethod: 'both',
+				category: 'template-classification-dialog',
+				action: 'change',
```

Comment:
Again, click might be more suitable here (matches your comment above). I'm also nor sure if there's a 'change' action predefined somewhere.